internal/venice-client-common/src/main/java/com/linkedin/venice/io/ZeroCopyByteArrayOutputStream.java

```diff
@@ -0,0 +1,20 @@
+package com.linkedin.venice.io;
+
+import java.io.ByteArrayOutputStream;
+import java.nio.ByteBuffer;
+
+
+public class ZeroCopyByteArrayOutputStream extends ByteArrayOutputStream {
+  public ZeroCopyByteArrayOutputStream(int size) {
+    super(size);
+  }
+
+  @Override
+  public synchronized byte[] toByteArray() {
+    return buf;
+  }
+
+  public synchronized ByteBuffer toByteBuffer() {
```

Comment:
Should we use this convenience method in `GzipCompressor::compress` since it's doing basically the same thing?

Reply:
Good catch, thanks! Updated PR to use `ZCBYOS::toByteBuffer` in `GzipCompressor::compress`